pkg/deploy/strategy/rolling/rolling.go

```diff
@@ -150,17 +151,25 @@ func (s *RollingDeploymentStrategy) Deploy(from *kapi.ReplicationController, to
 	//
 	// Related upstream issue:
 	// https://github.com/kubernetes/kubernetes/pull/7183
-	to, err = s.client.ReplicationControllers(to.Namespace).Get(to.Name)
+	err = wait.Poll(1*time.Second, 10*time.Second, func() (done bool, err error) {
+		existing, err := s.client.ReplicationControllers(to.Namespace).Get(to.Name)
+		if err != nil {
+			return false, fmt.Errorf("couldn't look up deployment %s: %s", deployutil.LabelForDeployment(to), err)
```

Comment:
Be careful because this is different than what you had before. With your custom retry func, you would retry on errors, here Poll will [exit](https://github.com/kubernetes/kubernetes/blob/0902f80f8b94fd23c66398118fcdf9e845c165b1/pkg/util/wait/wait.go#L73).